port/unix/omrsignal.c

```diff
@@ -502,17 +528,22 @@ omrsig_set_single_async_signal_handler(struct OMRPortLibrary *portLibrary, omrsi
 
 	if (0 != portlibSignalFlag) {
 		/* For non-zero portlibSignalFlag, check if only one signal bit is set. Otherwise, fail. */
-		if (!OMR_IS_ONLY_ONE_BIT_SET(portlibSignalFlag)) {
+		if (!OMR_IS_ONLY_ONE_BIT_SET((portlibSignalFlag & ~OMRPORT_SIG_FLAG_CONTROL_BITS_MASK))) {
```

Comment:
The problem is that `OMR_IS_ONLY_ONE_BIT_SET` does not properly use parentheses with its argument. Please fix that instead of adding parentheses here.

Reply:
amended.